blog/2024-01-28-reductstore-v1_8_0-released/index.md

```diff
@@ -0,0 +1,70 @@
+
+<!--more-->
+{/* truncate */}
+
+### Replication in ReductStore
+
+#### Append-Only Approach
+
+Currently, replication is append-only, meaning we only synchronize new records when they are written. As ReductStore is typically used as a large ring-buffer for blob data, it automatically removes old data when the disk quota is exceeded. Therefore, it doesn't make sense to replicate the remove operation on a remote instance where the disk space might be different.
+
+#### Edge/IoT Optimization
+
+Our goal is to create a storage solution tailored for IoT and Edge computing, and this consideration influenced our approach to replication.
+
+ReductStore's replication uses a transaction log to persistently store record IDs for replication. If a remote instance is unavailable, the database retains these records in its logs. This method allows for later synchronization when the remote instance is restored.
+
+Since an IoT device may not always have a public IP for various reasons, ReductStore implements a "push" model. In this model, a source instance writes data to a destination instance, eliminating the need for direct access to the source device.
+
+#### Conditional Replication
+
+ReductStore does more than just mirror data; it also provides conditions based on labels to filter data before sending it to another instance. This type of data reduction helps reduce traffic and extend storage time for important data.
+
+### Typical Use Cases
+
+We will soon provide detailed information about use cases for replication. For now, let's briefly discuss two scenarios where replication can help simplify your application.
+
+#### Local Replication
+
+In this use case, we have a computer vision application that detects product anomalies using images from a CV camera. The detection algorithm labels images as good or anomalous and stores them in the data bucket. We receive a few images per second, and the disk space is only sufficient to store a few days' worth of data. This amounts to a lot of images, and we would like to improve or validate our model with anomalous images. Unfortunately, anomalies occur very rarely, and we don't have enough images within our storage capacity.
+
+This is a perfect case for replication. We can create an additional bucket with a small quota and replicate only the anomalous images there. Depending on the quota chosen, it may store a few months' worth of data, which could be sufficient for our purposes.
+
+![Local Replication Example](./img/local-replication.webp)
+
+
+#### Remote Replication
+
+What if our device doesn't have a public IP, or the connection quality is too low for direct device interaction? We could address this by setting up a server with a ReductStore in our infrastructure and replicating the anomalous images from the edge device there.
+
```

Comment:
We could address this by setting up a server with a ReductStore database in our infrastructure and replicating the anomalous images from the edge device to the server.